### ISSUE TITLE
Remap categories before export for collections

### ISSUE THREAD

**Issue by grololo06** [ecotaxa/ecotaxa_dev#599]
#577 for collections, API only
@jiho , if you think about a re-mapping for the collections in #426, it's time to tell :)

**Comment by jiho** [comment]
Yes there will definitely be a remapping before export of a collection to EMODnet. Stating it there too.

**Comment by jiho** [comment]
And the remapping at collection level is independent from that of the underlying projects.

**Comment by grololo06** [comment]
OK this export mapping will be stored at collection level and accessible via an API endpoint.

**Comment by grololo06** [comment]
Correspondance b/w
    one category ID (classif_id) (assigned to at least one object in the collection) -> new category ID
e.g. part->crustacea becomes crustacea
For 2021 export: https://docs.google.com/spreadsheets/d/1WhTI6DzkL_J18Tgk8x0kSM5_KKsRt4MESuZHqTeKbbY/edit#gid=249061029

**Comment by jiho** [comment]
For the particular case of DwCA export, the interface should automatically propose to rename any non-worms taxon with a worms ancestor to that ancestor; the the user chooses to accept this or not. For example, both `female<Calanoida` and `part<Calanoida` should be proposed to be regrouped by default in `Calanoida` instead of thrown away and the user has to explicitly choose to throw `part<Calanoida` away. This would limit the amount of data thrown away.

**Comment by grololo06** [comment]
Renaming present issue to narrow to API-side of the work. On storage side, the plan is to use something like https://github.com/ecotaxa/ecotaxa_back/blob/master/py/DB/TaxoRecast.py . The "recast" term has been chosen as there are already too many "mappings" in EcoTaxa.

**Comment by grololo06** [comment]
https://github.com/ecotaxa/ecotaxa_front/issues/599#issuecomment-1018528797 implies that somehow the UI has an API endpoint for determining if a taxon is in WoRMS. So far there is `renm_id` field in returned calls, it's not specifically aimed at this, but could be probably.

**Comment by grololo06** [comment]
It looks like the remap should not take place for Occurences export, so refactoring is needed as 2 taxo "spaces" are needed, one for occurrences, one for computations.
